spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc

```diff
@@ -215,18 +215,120 @@ spring.cloud.stream.kafka.binder.configuration.security.protocol=SASL_SSL
-Here is an example of launching a Spring Cloud Stream application with SASL and Kerberos.
+
+Spring Cloud Stream supports passing JAAS configuration information to the application using a JAAS configuration file and using Spring Boot properties.
+
+===== Using JAAS configuration files
+
+The JAAS, and (optionally) krb5 file locations can be set for Spring Cloud Stream applications by using system properties.
+Here is an example of launching a Spring Cloud Stream application with SASL and Kerberos using a JAAS configuration file:
 
 [source]
 ----
- java -Djava.security.auth.login.config=/path.to/kafka_client_jaas.conf -jar log.jar \\
-   --spring.cloud.stream.kafka.binder.brokers=secure.server:9092 \\
-   --spring.cloud.stream.kafka.binder.zkNodes=secure.zookeeper:2181 \\
-   --spring.cloud.stream.bindings.input.destination=stream.ticktock \\
+ java -Djava.security.auth.login.config=/path.to/kafka_client_jaas.conf -jar log.jar \
+   --spring.cloud.stream.kafka.binder.brokers=secure.server:9092 \
+   --spring.cloud.stream.kafka.binder.zkNodes=secure.zookeeper:2181 \
+   --spring.cloud.stream.bindings.input.destination=stream.ticktock \
    --spring.cloud.stream.kafka.binder.clientConfiguration.security.protocol=SASL_PLAINTEXT
 ----
 
+===== Using Spring Boot properties
+
+As an alternative to having a JAAS configuration file, Spring Cloud Stream provides a mechanism for setting up the JAAS configuration for Spring Cloud Stream applications using Spring Boot properties.
+
+The following properties can be used for configuring the login context of the Kafka client.
+
+spring.cloud.stream.kafka.binder.jaas.loginModule::
+  The login module name. Not necessary to be set in normal cases.
++
+Default: `com.sun.security.auth.module.Krb5LoginModule`.
+spring.cloud.stream.kafka.binder.jaas.controlFlag::
+  The control flag of the login module.
++
+Default: `required`.
+spring.cloud.stream.kafka.binder.jaas.options::
+  Map with a key/value pair containing the login module options.
++
+Default: Empty map.
+
+If the application also requires access to a secured Zookeeper instance, e.g. if it needs to create and configure topics on the broker, the following properties can be set:
+
+spring.cloud.stream.kafka.binder.zkJaas.loginModule::
+  The login module name. Not necessary to be set in normal cases.
++
+Default: `com.sun.security.auth.module.Krb5LoginModule`.
+spring.cloud.stream.kafka.binder.zkJaas.controlFlag::
+  The control flag of the login module.
++
+Default: `required`.
+spring.cloud.stream.kafka.binder.zkJaas.options::
+  Map with a key/value pair containing the login module options.
++
+Default: Empty map.
+
+Here is an example of launching a Spring Cloud Stream application with SASL and Kerberos using Spring Boot configuration properties:
+
+[source]
+----
+ java -Djava.security.auth.login.config=/path.to/kafka_client_jaas.conf -jar log.jar \
```

Comment:
I have a feeling that in your tests you've still used `-Djava.security.auth.login.config` as using below throws error:
```
java -jar time-source-kafka-1.0.3.BUILD-SNAPSHOT.jar \
  --spring.cloud.stream.kafka.binder.brokers=sandbox:6667 \
  --spring.cloud.stream.kafka.binder.zkNodes=sandbox:2181 \
  --spring.cloud.stream.bindings.output.destination=ticktock1 \
  --spring.cloud.stream.kafka.binder.configuration.security.protocol=SASL_PLAINTEXT \
  --spring.cloud.stream.kafka.binder.jaas.options.useKeyTab=true \
  --spring.cloud.stream.kafka.binder.jaas.options.storeKey=true \
  --spring.cloud.stream.kafka.binder.jaas.options.keyTab=/etc/security/keytabs/scdf.service.keytab \
  --spring.cloud.stream.kafka.binder.jaas.options.principal=scdf/sandbox.hortonworks.com@HORTONWORKS.COM \
  --spring.cloud.stream.kafka.binder.zkJaas.options.useKeyTab=true \
  --spring.cloud.stream.kafka.binder.zkJaas.options.storeKey=true \
  --spring.cloud.stream.kafka.binder.zkJaas.options.keyTab=/etc/security/keytabs/scdf.service.keytab \
  --spring.cloud.stream.kafka.binder.zkJaas.options.principal=scdf/sandbox.hortonworks.com@HORTONWORKS.COM \
  --server.port=0
```
```
Caused by: org.springframework.cloud.stream.binder.BinderException: Cannot initialize binder:
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.getPartitionsForTopic(KafkaMessageChannelBinder.java:524)
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.createProducerDestinationIfNecessary(KafkaMessageChannelBinder.java:230)
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.createProducerDestinationIfNecessary(KafkaMessageChannelBinder.java:93)
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindProducer(AbstractMessageChannelBinder.java:98)
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindProducer(AbstractMessageChannelBinder.java:53)
	at org.springframework.cloud.stream.binder.AbstractBinder.bindProducer(AbstractBinder.java:148)
	at org.springframework.cloud.stream.binding.ChannelBindingService.bindProducer(ChannelBindingService.java:113)
	at org.springframework.cloud.stream.binding.BindableProxyFactory.bindOutputs(BindableProxyFactory.java:208)
	at org.springframework.cloud.stream.binding.OutputBindingLifecycle.start(OutputBindingLifecycle.java:57)
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:173)
	... 20 more
Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:321)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:171)
	at org.springframework.kafka.core.DefaultKafkaProducerFactory.createKafkaProducer(DefaultKafkaProducerFactory.java:138)
	at org.springframework.kafka.core.DefaultKafkaProducerFactory.createProducer(DefaultKafkaProducerFactory.java:130)
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder$4.doWithRetry(KafkaMessageChannelBinder.java:510)
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder$4.doWithRetry(KafkaMessageChannelBinder.java:503)
	at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:263)
	at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:154)
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.getPartitionsForTopic(KafkaMessageChannelBinder.java:503)
	... 29 more
Caused by: org.apache.kafka.common.KafkaException: java.lang.IllegalArgumentException: You must pass java.security.auth.login.config in secure mode.
	at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:74)
	at org.apache.kafka.common.network.ChannelBuilders.create(ChannelBuilders.java:60)
	at org.apache.kafka.clients.ClientUtils.createChannelBuilder(ClientUtils.java:79)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:271)
	... 37 more
Caused by: java.lang.IllegalArgumentException: You must pass java.security.auth.login.config in secure mode.
	at org.apache.kafka.common.security.kerberos.Login.login(Login.java:289)
	at org.apache.kafka.common.security.kerberos.Login.<init>(Login.java:104)
	at org.apache.kafka.common.security.kerberos.LoginManager.<init>(LoginManager.java:44)
	at org.apache.kafka.common.security.kerberos.LoginManager.acquireLoginManager(LoginManager.java:85)
	at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:55)
	... 40 more
```
as there is yet another property check at https://github.com/apache/kafka/blob/0.9.0/clients/src/main/java/org/apache/kafka/common/security/kerberos/Login.java#L287
```
    private synchronized LoginContext login(final String loginContextName) throws LoginException {
        String jaasConfigFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM);
        if (jaasConfigFile == null) {
            throw new IllegalArgumentException("You must pass " + JaasUtils.JAVA_LOGIN_CONFIG_PARAM + " in secure mode.");
        }

```